runtime/src/commonMain/kotlin/app/cash/sqldelight/EnumColumnAdapter.kt

```diff
@@ -15,17 +15,20 @@
  */
 package app.cash.sqldelight
 
+import kotlin.enums.EnumEntries
+import kotlin.enums.enumEntries
+
 /** A [ColumnAdapter] which maps the enum class `T` to a string in the database. */
 class EnumColumnAdapter<T : Enum<T>> @PublishedApi internal constructor(
-  private val enumValues: Array<out T>,
+  private val enumEntries: EnumEntries<out T>,
 ) : ColumnAdapter<T, String> {
-  override fun decode(databaseValue: String): T = enumValues.first { it.name == databaseValue }
+  override fun decode(databaseValue: String): T = enumEntries.first { it.name == databaseValue }
```

Comment:
This now allocates an iterator on each invocations, whereas before it used an index-based loop. The `EnumEntries` type does not implement `RandomAccess` (filed [KT-74444](https://youtrack.jetbrains.com/issue/KT-74444)), but it's still probably safe to use index-based iteration here.